packages/opentelemetry-resource-detector-aws/src/detectors/AwsEksDetector.ts

```diff
@@ -0,0 +1,227 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Detector,
+  Resource,
+  CONTAINER_RESOURCE,
+  K8S_RESOURCE,
+  ResourceDetectionConfigWithLogger,
+} from '@opentelemetry/resources';
+import * as https from 'https';
+import * as fs from 'fs';
+import * as util from 'util';
+
+/**
+ * The AwsEksDetector can be used to detect if a process is running in AWS Elastic
+ * Kubernetes and return a {@link Resource} populated with data about the Kubernetes
+ * plugins of AWS X-Ray. Returns an empty Resource if detection fails.
+ *
+ * See https://docs.amazonaws.cn/en_us/xray/latest/devguide/xray-guide.pdf
+ * for more details about detecting information for Elastic Kubernetes plugins
+ */
+
+export class AwsEksDetector implements Detector {
+  readonly K8S_SVC_URL = 'kubernetes.default.svc';
+  readonly K8S_TOKEN_PATH =
+    '/var/run/secrets/kubernetes.io/serviceaccount/token';
+  readonly K8S_CERT_PATH =
+    '/var/run/secrets/kubernetes.io/serviceaccount/ca.crt';
+  readonly AUTH_CONFIGMAP_PATH =
+    '/api/v1/namespaces/kube-system/configmaps/aws-auth';
+  readonly CW_CONFIGMAP_PATH =
+    '/api/v1/namespaces/amazon-cloudwatch/configmaps/cluster-info';
+  readonly CONTAINER_ID_LENGTH = 64;
+  readonly DEFAULT_CGROUP_PATH = '/proc/self/cgroup';
+  readonly MILLISECOND_TIME_OUT = 2000;
+
+  private static readFileAsync = util.promisify(fs.readFile);
+  private static fileAccessAsync = util.promisify(fs.access);
+
+  /**
+   * The AwsEksDetector can be used to detect if a process is running in AWS
+   * Eks and returns a promise containing a {@link Resource}
+   * populated with instance metadata. Returns a promise containing an
+   * empty {@link Resource} if the connection to kubernetes process
+   * or aws config maps fails
+   * @param config The resource detection config with a required logger
+   */
+  async detect(config: ResourceDetectionConfigWithLogger): Promise<Resource> {
+    try {
+      await AwsEksDetector.fileAccessAsync(this.K8S_TOKEN_PATH);
+      const k8scert = await AwsEksDetector.readFileAsync(this.K8S_CERT_PATH);
+
+      if (!this._isEks(config, k8scert)) {
+        return Resource.empty();
+      }
+
+      const containerId = await this._getContainerId(config);
+      const clusterName = await this._getClusterName(config, k8scert);
+
+      return !containerId && !clusterName
+        ? Resource.empty()
+        : new Resource({
+            [K8S_RESOURCE.CLUSTER_NAME]: clusterName || '',
+            [CONTAINER_RESOURCE.ID]: containerId || '',
+          });
+    } catch (e) {
+      config.logger.warn('Process is not running on K8S', e);
+      return Resource.empty();
+    }
+  }
+
+  /**
+   * Attempts to make a connection to AWS Config map which will
+   * determine whether the process is running on an Eks
```

Comment:
if this is a shortcut `Eks` shouldn't this be with capital letters ? `EKS` or simply use the full name ?